sdk/core/azure-json/src/test/java/com/azure/json/contract/JsonReaderContractTests.java

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.json.contract;
+
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Tests the contract of {@link JsonReader}.
+ * <p>
+ * All implementations of {@link JsonReader} must create a subclass of this test class and pass all tests as they're
+ * written to be considered an acceptable implementation.
+ * <p>
+ * Each test will only create a single instance of {@link JsonReader} to simplify the usage of
+ * {@link #getJsonReader(String)}.
+ */
+public abstract class JsonReaderContractTests {
```

Comment:
would be nice to use extensive run autorest testing with the latest azure-json at least before each release, but ideally on each change.